src/host/ft_host/API_OutputTests.cpp

```diff
@@ -159,15 +159,16 @@ void OutputTests::WriteConsoleOutputWOutsideBuffer()
     // Call the API and confirm results.
 
     //  move outside in X and Y directions
-    auto shiftedRegion = region;
-    shiftedRegion.Left += bufferSize.X;
-    shiftedRegion.Right += bufferSize.X;
-    shiftedRegion.Top += bufferSize.Y;
-    shiftedRegion.Bottom += bufferSize.Y;
-
-    auto affected = shiftedRegion;
+    auto affected = region;
+    affected.Left += bufferSize.X;
+    affected.Right += bufferSize.X;
+    affected.Top += bufferSize.Y;
+    affected.Bottom += bufferSize.Y;
+    auto expected = affected;
+    expected.Right = expected.Left - 1;
+    expected.Bottom = expected.Top - 1;
     VERIFY_WIN32_BOOL_SUCCEEDED(WriteConsoleOutputW(consoleOutputHandle, buffer.data(), regionDimensions, regionOrigin, &affected));
-    VERIFY_ARE_EQUAL(shiftedRegion, affected);
+    VERIFY_ARE_EQUAL(expected, affected);

```

Comment:
This test ironically asserted a fairly grave bug in our `WriteConsoleOutput` implementation. MSDN clearly states:
> If the rectangle specified by `lpWriteRegion` lies completely outside the boundaries of the console screen buffer, [...] no data is written. In this case, the function returns [...] such that the `Right` member is less than the `Left`, or the `Bottom` member is less than the `Top`.

Meanwhile this test asserted that we input return the rectangle unmodified.

This happens because the only thing we test is whether the `lpWriteRegion` has a negative size:
https://github.com/microsoft/terminal/blob/ec9289288e6581b8d1cbb5f7d334422de4dee07a/src/host/directio.cpp#L479-L484

This PR fixes that. (It had to because this bug caused the `ReadConsoleOutput` + `WriteConsoleOutput` console -> VT translation to not work.)